tests/apps/tests.py

```diff
@@ -528,3 +531,112 @@ def test_multiple_paths_explicit_path(self):
             with self.settings(INSTALLED_APPS=["nsapp.apps.NSAppConfig"]):
                 app_config = apps.get_app_config("nsapp")
                 self.assertEqual(app_config.path, self.app_path)
+
+
+class QueryPerformingAppTests(SimpleTestCase):
+    databases = {"default", "other"}
+
+    expected_message = (
+        "Accessing the database during app initialization is discouraged."
+    )
+
+    def tearDown(self) -> None:
```

Comment:
Please remove type annotations. We don't currently use them in Django.

```suggestion
    def tearDown(self):
```